app/models/client.rb

```diff
@@ -312,18 +355,19 @@ def make_unavailable_in_all_routes
   end
 
   def unavailable(permanent:false, contact_id:nil)
-    active_match = client_opportunity_matches.active.first
     Client.transaction do
-      if active_match.present?
-        opportunity = active_match.opportunity
-        active_match.canceled!
-        MatchEvents::Parked.create!(
-            match_id: active_match.id,
-            contact_id: contact_id
-        )
-        opportunity.update(available_candidate: true)
-        Matching::RunEngineJob.perform_later
+
+      active_matches do |active_match|
+        active_match.canceled!(contact_id)
```

Comment:
`ClientOpportunityMatch#canceled!` takes a named parameter `contact_id:`, this will not work as written.

Reply:
ok.  It seemed to work in testing, but I will make the change.

Reply:
weird.  It seemed to worked in testing, but ok.